### PR TITLE
Enable string format operations with PoolArrays

### PR DESCRIPTION
**Pull request by lupoDharkael**
solves #27985

image

### CODE REVIEW

File: modules/gdscript/gdscript_parser.cpp [hunk 1]
```diff
@@ -5919,7 +5919,7 @@ GDScriptParser::DataType GDScriptParser::_get_operation_type(const Variant::Oper
 	if (b_type == Variant::INT || b_type == Variant::REAL) {
 		Variant::evaluate(Variant::OP_ADD, b, 1, b, r_valid);
 	}
-	if (a_type == Variant::STRING && b_type != Variant::ARRAY) {
+	if (a_type == Variant::STRING && b_type != Variant::ARRAY && b_type != Variant::POOL_STRING_ARRAY && b_type != Variant::POOL_INT_ARRAY && b_type != Variant::POOL_BYTE_ARRAY && b_type != Variant::POOL_COLOR_ARRAY && b_type != Variant::POOL_REAL_ARRAY && b_type != Variant::POOL_VECTOR2_ARRAY && b_type != Variant::POOL_VECTOR3_ARRAY) {
```
Comment (lupoDharkael) [inline modules/gdscript/gdscript_parser.cpp]: Is there another way of doing this? it feels bad to write that number of conditions. I know they are needed in order to pass the operation evaluation of the script parsing.

Comment (lupoDharkael) [reply modules/gdscript/gdscript_parser.cpp]: ok, I've been checking the variant.h file and I've seen the `is_array()` method, I'll use it instead.

**Comment by aaronfranke** [comment]
@lupoDharkael Is this still desired? If so, it needs to be rebased on the latest master branch. While there are no conflicts, this code wouldn't work if merged, since Pool*Arrays have been removed and replaced with Packed*Arrays.

**Comment by aaronfranke** [comment]
Closing as abandoned by the author. If this is still desired, feel free to open a new PR, redesigned to work with PackedArrays.

**Pull request closed without merging**